engines/dragons/actor.cpp

```diff
@@ -0,0 +1,794 @@
+	int pathPointsIndex = 0;
+	while (!canWalkLine(actorX1, actorY1, tempDestX1, tempDestY1, flags) && pathPointsIndex < kPathPointsCount) {
+		int foundPointIndex = pathfindingFindClosestPoint(actorX1, actorY1, tempDestX1, tempDestY1, flags, pathPointProcessed);
+		if (foundPointIndex < 0) {
+			if (wasAlreadyWalking) {
+				stopWalk();
+			}
+			return false;
+		}
+		pathPointProcessed[foundPointIndex] = true;
+		const Common::Point pt = getEngine()->_scene->getPoint(foundPointIndex);
+		tempDestX1 = pt.x;
+		tempDestY1 = pt.y;
+		if (pathPointsIndex >= 2) {
+			const Common::Point prevPt = getEngine()->_scene->getPoint(walkPointsTbl[pathPointsIndex - 2]);
+			if (canWalkLine(pt.x, pt.y, prevPt.x, prevPt.y, flags)) {
+				--pathPointsIndex;
+			}
+		} else if (pathPointsIndex == 1) {
+			if (canWalkLine(pt.x, pt.y, destX, destY, flags)) {
+				--pathPointsIndex;
+			}
+		}
+		walkPointsTbl[pathPointsIndex] = foundPointIndex;
+		++pathPointsIndex;
+	}
+
+	// Direction/post-processing
+	if (xorflagsl != 0) {
+		uint destDistance = calcDistance(destX, destY, tempDestX1, tempDestY1);
+		uint sourceDistance = calcDistance(actorX1, actorY1, destX, destY);
+		if (sourceDistance < 625 && ((actorX1 == destX && actorY1 == destY) || (sourceDistance < destDistance))) {
+			int newDirection;
+			int dx = origDestX - actorX1;
+			int dy = origDestY - actorY1;
+			if (dx != 0) {
+				int slope = dy / dx;
+				if (slope == 0) {
+					newDirection = (dx < 1) ? 4 : 0;
+				} else if (slope > 0) {
+					newDirection = (dx <= 0) ? 3 : 7;
+				} else {
+					newDirection = (dx <= 0) ? 5 : 1;
+				}
+			} else {
+				newDirection = (dy <= 0)  ? 2 : 6;
+			}
+			_sequenceID2 = newDirection;
+			if (wasAlreadyWalking) {
+				stopWalk();
+			}
+			return false;
+		}
+	}
+
+	walkPointsIndex = pathPointsIndex - 1;
+	if (pathPointsIndex == 0) {
+		_walkDestX = tempDestX1;
+		_walkDestY = tempDestY1;
+		_finalWalkDestX = -1;
+		_finalWalkDestY = -1;
+	} else {
+		const Common::Point pt = getEngine()->_scene->getPoint(walkPointsTbl[walkPointsIndex]);
+		_walkDestX = pt.x;
+		_walkDestY = pt.y;
+	}
+	int direction = startMoveToPoint(_walkDestX, _walkDestY);
+	if (direction != -1 && !isFlagSet(ACTOR_FLAG_800)) {
+		_sequenceID2 = direction;
+	}
+	if (_sequenceID != _sequenceID2 + 8 && !isFlagSet(ACTOR_FLAG_800)) {
+		updateSequence(_sequenceID2 + 8);
+	}
+	setFlag(ACTOR_FLAG_10);
+	return true;
+}
+
+void Actor::stopWalk() {
+	clearFlag(Dragons::ACTOR_FLAG_10);
+	walkPointsIndex = 0;
+	_walkDestX = x_pos;
+	_walkDestY = y_pos;
+	_finalWalkDestX = -1;
+	_finalWalkDestY = -1;
+	setFlag(Dragons::ACTOR_FLAG_4);
+
+	if (_flags & Dragons::ACTOR_FLAG_200) {
+		clearFlag(Dragons::ACTOR_FLAG_800);
+	}
+}
+
+void Actor::waitUntilFlag4IsSet() {
+	while(!isFlagSet(ACTOR_FLAG_4)) {
+		getEngine()->waitForFrames(1);
+	}
+}
+
+void Actor::waitUntilFlag8IsSet() {
+	if (_flags & Dragons::ACTOR_FLAG_8) {
+		return;
+	}
+
+	while(!(_flags & Dragons::ACTOR_FLAG_8)) {
```

Comment:
code formatting. Please search `while(` and replace it. I'll not spam with future cases like this.